plugin.py

```diff
@@ -93,6 +107,56 @@ def install_or_update(cls) -> None:
             shutil.rmtree(cls.basedir(), ignore_errors=True)
             raise
 
+    def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
+        if not command["command"].startswith("rust-analyzer"):
+            return False
+        cargo_commands = []
+        for c in command["arguments"]:
+            if c["kind"] == "cargo":
+                cargo_commands.append(c)
```

Comment:
I've set up rust here to check and gotten this error on clicking annotation that said `10 implementations`:

```
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1488, in run_
    return self.run(edit, **args)
  File "/.../LSP/plugin/execute_command.py", line 49, in run
    session.execute_command(params, progress=True).then(handle_response)
  File "/.../LSP/plugin/core/sessions.py", line 1065, in execute_command
    if self._plugin.on_pre_server_command(command, lambda: resolve(None)):
  File "/.../LSP-rust-analyzer/plugin.py", line 115, in on_pre_server_command
    if c["kind"] == "cargo":
TypeError: string indices must be integers
```

So that's another reason for catching only specific commands.
Also, your assumptions that `kind` property always exists is not holding here. Either because you are not supposed to handle this command (`rust-analyzer.showReferences`) or because the property is not guaranteed to exists.

Reply:
Okay lemme be more specific